packages/sourcecred/src/api/plugin.js

```diff
@@ -22,9 +22,210 @@ export interface Plugin {
+  |}>,
+  score: number,
+|};
+
+export const exampleDiscordContribution: UnscoredContribution = {
+  id: "1234",
+  plugin: "Discord",
+  type: "Message",
+  participants: [
+    {id: "ASDVERW342tFD", shares: 1},
+    {id: "SD533FEfdsdrG", shares: 3},
+  ],
+  equation: {
+    type: "MULTIPLY",
+    description: "contribution attributes",
+    factors: [{key: "channel", value: "12345"}],
+    equationFactors: [
+      {
+        type: "ADD",
+        description: "reactions",
+        factors: [],
+        equationFactors: [
+          {
+            type: "MULTIPLY",
+            description: "reaction attributes",
+            factors: [
+              {key: "emoji", value: "asdf"},
+              {key: "role", value: "123445"},
+            ],
+            equationFactors: [],
+          },
+        ],
+      },
+    ],
+  },
+};
+
+export const exampleDiscordWeightConfig: WeightConfig = [
+  {
+    key: "channel",
+    default: 2,
+    values: [
+      {value: "12345", weight: 3},
+      {value: "45678", weight: 1},
+    ],
+  },
+  {
+    key: "emoji",
+    default: 1,
+    values: [],
+  },
+  {
+    key: "role",
+    default: 2,
+    values: [],
+  },
+];
+
+const evaluate: (Equation, WeightConfig) => number = (
+  equation,
+  weightConfig
+) => {
+  let equationFactorResult, factorResult, result;
+  switch (equation.type) {
+    case "MULTIPLY":
+      equationFactorResult = equation.equationFactors.reduce(
+        (accumulator, factor) => {
+          return (
+            evaluate(factor, weightConfig) *
+            (accumulator === undefined ? 1 : accumulator)
+          );
+        },
+        1
+      );
+      factorResult = equation.factors.reduce((accumulator, factor) => {
+        const config = weightConfig.find((x) => x.key === factor.key);
+        if (config === undefined) throw "Unexpected factor key " + factor.key;
+        return (
+          (config.values.find((x) => x.value === factor.value)?.weight ||
+            config.default) * (accumulator === undefined ? 1 : accumulator)
+        );
+      }, 1);
+      if (equationFactorResult === undefined || factorResult === undefined) {
+        throw "not supposed to happen";
+      }
+      result = factorResult * equationFactorResult;
+      equation.score = result;
+      return result;
+
+    case "ADD":
+      equationFactorResult = equation.equationFactors.reduce(
+        (accumulator, factor) => {
+          return (
+            evaluate(factor, weightConfig) +
+            (accumulator === undefined ? 1 : accumulator)
+          );
+        },
+        0
+      );
+      factorResult = equation.factors.reduce((accumulator, factor) => {
+        const config = weightConfig.find((x) => x.key === factor.key);
+        if (config === undefined) throw "Unexpected factor key " + factor.key;
+        return (
+          (config.values.find((x) => x.value === factor.value)?.weight ||
+            config.default) + (accumulator === undefined ? 1 : accumulator)
```

Comment:
```suggestion
          (config.values.find((x) => x.value === factor.value)?.weight ??
            config.default) + (accumulator === undefined ? 1 : accumulator)
```

When `value.weight === 0`, won't it use the default value?

Reply:
ah good catch